backend-services/src/data_service/schema/entities.py

```diff
@@ -0,0 +1,65 @@
+from typing import TypeAlias
+
+from odmantic import Model
+from pydantic import BaseModel
+
+from common.types import WalletAddress
+
+from datetime import date
+
+class Dataset(Model):
+    """
+    An address on the ledger dataseted by the user.
+    """
+
+    wallet_id: WalletAddress
+    data_pool_id: str
+    name: str
+    description: str
+    length: int
+    created: date
```

Comment:
replace all the `date` with `datetime`

Turns out that. PyMongo doesn't support saving date instances. The server doesn't have a type for dates without times, so there would have to be some convention used to save dates without times.